src/main/java/hudson/plugins/promoted_builds/Promotion.java

```diff
@@ -87,6 +90,13 @@ public Status getStatus() {
         return getTarget().getAction(PromotedBuildAction.class).getPromotion(getParent().getName());
     }
 
+	public String getIcon() {
+		return icon;
+	}
+
+	public void setIcon(String icon) {
```

Comment:
Could you explain this a bit more? In a concurrent environment. Do you mean the implementation should be similar to how the description field in the Run class is set?

something like below?
```
   protected volatile String icon;

   public String getIcon() {
     return icon;
   }

   public void setIcon(String description) throws IOException {
      checkPermission(UPDATE);
      this.icon= icon;
      save();
   }
```